internal/jujuapi/jimm_unit_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2025 Canonical.
+
+package jujuapi_test
+
+import (
+	"context"
+
+	"github.com/juju/names/v5"
+	gc "gopkg.in/check.v1"
+
+	"github.com/canonical/jimm/v3/internal/openfga"
+	"github.com/canonical/jimm/v3/internal/testutils/jimmtest/mocks"
+	apiparams "github.com/canonical/jimm/v3/pkg/api/params"
+)
+
+type jimmUnitTestSuite struct{}
+
+var _ = gc.Suite(&jimmUnitTestSuite{})
+
+func (s *jimmSuite) TestPrepareModelMigration_UnauthorizedUser(c *gc.C) {
+	ctx := context.Background()
+
+	jujuManager := mocks.JujuManager{
+		PrepareModelMigration_: func(ctx context.Context, user *openfga.User, modelUUID, targetControllerName string, userMapping map[string]string) error {
+			return nil
+		},
+	}
+	root := newTestControllerRoot(jujuManager, "alice@canonical.com", false)
+
+	err := root.PrepareModelMigration(ctx, apiparams.PrepareModelMigrationRequest{})
+
+	c.Assert(err, gc.ErrorMatches, "unauthorized")
+}
+
+func (s *jimmSuite) TestPrepareModelMigration_InvalidModelTag(c *gc.C) {
+	ctx := context.Background()
+
+	jujuManager := mocks.JujuManager{
+		PrepareModelMigration_: func(ctx context.Context, user *openfga.User, modelUUID, targetControllerName string, userMapping map[string]string) error {
+			return nil
+		},
+	}
+	root := newTestControllerRoot(jujuManager, "alice@canonical.com", true)
+
+	err := root.PrepareModelMigration(ctx, apiparams.PrepareModelMigrationRequest{
+		ModelTag: "blah",
+	})
+
+	c.Assert(err, gc.ErrorMatches, "invalid model tag")
+}
+
+func (s *jimmSuite) TestPrepareModelMigration_InvalidControllerName(c *gc.C) {
+	ctx := context.Background()
+
+	jujuManager := mocks.JujuManager{
+		PrepareModelMigration_: func(ctx context.Context, user *openfga.User, modelUUID, targetControllerName string, userMapping map[string]string) error {
+			return nil
+		},
+	}
+	root := newTestControllerRoot(jujuManager, "alice@canonical.com", true)
+
+	err := root.PrepareModelMigration(ctx, apiparams.PrepareModelMigrationRequest{
+		ModelTag:             names.NewModelTag("5650ac3f-8332-437f-874f-089e0e447e7f").String(),
+		TargetControllerName: "---bad wolf---",
+	})
+
+	c.Assert(err, gc.ErrorMatches, "invalid controller name")
+}
+
+func (s *jimmSuite) TestPrepareModelMigration_InvalidUserMapping(c *gc.C) {
+	ctx := context.Background()
+
+	jujuManager := mocks.JujuManager{
+		PrepareModelMigration_: func(ctx context.Context, user *openfga.User, modelUUID, targetControllerName string, userMapping map[string]string) error {
+			return nil
+		},
+	}
```

Comment:
You could delete this and the others since it should never reach the the point where it calls the PrepareModelMigration method.